File: app/classes/lobby.py

class Player:
    def __init__(self, username, points, is_ready):
        self.username = username
        self.points = points
        self.is_ready = is_ready

    def to_dict(self):
        return {
            'username': self.username,
            'points': self.points,
            'is_ready': self.is_ready
        }


class Lobby:

    def __init__(self, turns: int = 10) -> None:
        self.turns = turns
        self.players = {}

    def to_dict(self):
        return {
            'turns': self.turns,
            'players': {
                username: player.to_dict() for username, player
                in self.players.items()
            }
        }

File: app/static/js/lobby.js
import * as SocketModule from './socket.js'

document.addEventListener('DOMContentLoaded', () => {

    const leaveButton = document.getElementById('leave');
    const toggleReadyButton = document.getElementById('toggleready');
    const playerContainer = document.getElementById('player-container');

    SocketModule.render_lobby_socket(renderLobbyContainer);

    function renderLobbyContainer(lobby) {
        console.log('diocane');
        const players = lobby.players;
        playerContainer.innerHTML = '';

        for (const [key, value] of Object.entries(players)) {
            const playerElement = document.createElement('div');
            playerElement.id = key;
            playerElement.innerHTML = `<h2>${key}</h2>`;
            playerElement.style.backgroundColor = value.is_ready ? 'green' : 'red';
            playerContainer.appendChild(playerElement);
        }
    }


    leaveButton.addEventListener('click', () => {
        const urlParams = new URLSearchParams(window.location.search);
        const username = urlParams.get('username');
        const lobbyName = urlParams.get('lobbyName');
        SocketModule.leave_lobby(username, lobbyName);
        window.location.href = `/`;
    });

    toggleReadyButton.addEventListener('click', () => {
        const urlParams = new URLSearchParams(window.location.search);
        const username = urlParams.get('username');
        const lobbyName = urlParams.get('lobbyName');
        SocketModule.toggle_ready(username, lobbyName);
    });


});